controller/instance_manager_controller.go

```diff
@@ -1702,34 +1701,52 @@ func (imc *InstanceManagerController) deleteOrphans(im *longhorn.InstanceManager
 	return nil
 }
 
-func (imc *InstanceManagerController) isOrphanDeletable(orphan *longhorn.Orphan, autoDelete bool, instanceMap instanceProcessMap, isInstanceScheduledOnNode func(node, instance string) (bool, error)) (isDeletable bool, err error) {
-	if autoDelete {
-		return true, nil
+func (imc *InstanceManagerController) deleteOrphan(orphan *longhorn.Orphan, resourceDeletable bool) error {
+	if orphan.Spec.ResourceDeletable != resourceDeletable {
+		orphan.Spec.ResourceDeletable = resourceDeletable
+		_, err := imc.ds.UpdateOrphan(orphan)
+		switch {
+		case datastore.ErrorIsNotFound(err):
+			return nil
+		case err != nil:
+			return errors.Wrapf(err, "failed to update resource deletable on orphan %v before deletion", orphan.Name)
+		}
 	}
 
+	err := imc.ds.DeleteOrphan(orphan.Name)
+	switch {
+	case datastore.ErrorIsNotFound(err):
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to delete orphan %v", orphan.Name)
+	}
+	return nil
+}
+
+func (imc *InstanceManagerController) areOrphanAndResourceDeletable(orphan *longhorn.Orphan, imTerminating, crAutoDelete bool, instanceMap instanceProcessMap, isInstanceScheduledOnNode func(node, instance string) (bool, error)) (isOrphanCRDeletable, isOrphanResourceDeletable bool, err error) {
```

Comment:
Suggest separating this into two methods, `canDeleteOrphan` and `canDeleteOrphanedInstance`, to improve maintainability.

Reply:
`canDeleteOrphanedInstance` is removed.